### ISSUE TITLE
Activation with "softmax" leads to segfault

### ISSUE THREAD

**Issue by simleo** [deephealthproject/eddl#231]
This code that used to work up to v0.8a now (v0.8.1a) leads to a segmentation fault:

```c++
layer out = Activation(Dense(l, num_classes), "softmax");
```

The workaround is to replace it with:

```c++
layer out = Softmax(Dense(l, num_classes));
```

I don't know if the intention is that the old way should still work: if so, it should be fixed. If not, it should raise some exception, rather than resulting in a segmentation fault.

Note that the old idiom still shows up in [some examples](https://github.com/deephealthproject/eddl/blob/7995de5efb3d48d9d2e22609b7c42fd7bb83281c/examples/onnx/nn/1_mnist/4_mnist_mlp_da.cpp#L54).

**Comment by RParedesPalacios** [comment]
Simone, thanks, not sure what happens, Salva was working on that, let's see....

**Comment by salvacarrion** [comment]
I've fixed the examples in #232
